comms/db/queries/get_chats.go

```diff
@@ -42,23 +52,49 @@ func UserChat(q db.Queryable, ctx context.Context, arg ChatMembershipParams) (Us
 
 // Get all chats (with user-specific details) for the given user
 const userChats = `
+WITH last_message_per_chat AS (
+	SELECT ciphertext, chat_id
+	FROM (
+		SELECT RANK() OVER (PARTITION BY chat_message.chat_id ORDER BY chat_message.created_at DESC, chat_message.message_id) AS chat_rank,
+		chat_message.ciphertext,
+		chat_message.chat_id
+		FROM chat_message
+		JOIN chat_member on chat_member.chat_id = chat_message.chat_id
+		WHERE chat_member.user_id = $1
+	) ranked_message
+	WHERE chat_rank = 1
+)
 SELECT
   chat.chat_id,
   chat.created_at, 
+	last_message_per_chat.ciphertext AS last_message,
   chat.last_message_at,
   chat_member.invite_code,
   chat_member.last_active_at,
   chat_member.unread_count,
   chat_member.cleared_history_at
 FROM chat_member
 JOIN chat ON chat.chat_id = chat_member.chat_id
-WHERE chat_member.user_id = $1 AND (chat_member.cleared_history_at IS NULL OR chat.last_message_at > chat_member.cleared_history_at)
+LEFT JOIN last_message_per_chat ON last_message_per_chat.chat_id = chat_member.chat_id
+WHERE chat_member.user_id = $1
+	AND chat.created_at < $3
```

Comment:
I think pagination should be by `last_message_at`?

Reply:
last_message_at is the timestamp of the latest message

Reply:
Yep - I think that's how we should paginate because that's what we order by

Reply:
ah right

Reply:
https://github.com/AudiusProject/audius-protocol/pull/4616/commits/156f05822d979e3ba4c424a8dc1a6f0e85193a7a!